pkg/gatherers/clusterconfig/dvo_metrics.go

```diff
@@ -0,0 +1,100 @@
+package clusterconfig
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"regexp"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/util/flowcontrol"
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/insights-operator/pkg/record"
+	"github.com/openshift/insights-operator/pkg/utils"
+	"github.com/openshift/insights-operator/pkg/utils/marshal"
+)
+
+const (
+	dvoNamespace = "deployment-validation-operator"
+	// dvo_metrics_endpoint = "http://deployment-validation-operator-metrics.deployment-validation-operator.svc:8383"
+)
+
+var (
+	dvoMetricsServiceNameRegex = regexp.MustCompile(`\bmetrics\b`)
+	dvoMetricsPrefix           = []byte("deployment_validation_operator_")
+)
+
+func (g *Gatherer) GatherDVOMetrics(ctx context.Context) ([]record.Record, []error) {
+	// apiURL, err := url.Parse(dvo_metrics_endpoint)
+	// if err != nil {
+	// 	return nil, []error{err}
+	// }
+	// metricsRESTClient, err := rest.NewRESTClient(apiURL, "/", rest.ClientContentConfig{}, g.gatherKubeConfig.RateLimiter, http.DefaultClient)
+	// if err != nil {
+	// 	klog.Warningf("Unable to load metrics client, no metrics will be collected: %v", err)
+	// 	return nil, nil
+	// }
+	gatherKubeClient, err := kubernetes.NewForConfig(g.gatherProtoKubeConfig)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	return gatherDVOMetrics(ctx, gatherKubeClient.CoreV1(), g.gatherKubeConfig.RateLimiter)
+}
+
+func gatherDVOMetrics(ctx context.Context, coreClient corev1client.CoreV1Interface, rateLimiter flowcontrol.RateLimiter) ([]record.Record, []error) {
+	serviceList, err := coreClient.Services(dvoNamespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	allDVOMetricsLines := []byte{}
+
+	for _, service := range serviceList.Items {
+		if !dvoMetricsServiceNameRegex.MatchString(service.Name) {
+			continue
+		}
+		for _, port := range service.Spec.Ports {
```

Comment:
Hmmm this is interesting. I didn't think about the option of service with more ports specified....Maybe it will be slightly better to expect (i.e fail otherwise) that there's only one port specified than doing/trying to do more requests otherwise. WDYT?

Reply:
I don't really know. Is it a problem to make more than just one request? I'm just thinking there could be some kind of an odd hybrid metrics service that exposes multiple ports instead of having multiple endpoints, or something wild like that.